bridges/scalajs-1/src/main/scala/bloop/scalajs/JsBridge.scala

```diff
@@ -51,6 +53,41 @@ object JsBridge {
       }
     override def trace(t: => Throwable): Unit = logger.trace(t)
   }
+  private object ScalaJSLinker {
+    private val cache = TrieMap.empty[Path, WeakReference[(JsConfig, Linker)]]
```

Comment:
`WeakReference`s are not the best choice for memory-sensitive caches. Use a [`SoftReference`](https://docs.oracle.com/javase/8/docs/api/java/lang/ref/SoftReference.html) instead.